sdk/servicebus/service-bus/src/serializers/queueResourceSerializer.ts

```diff
@@ -310,11 +320,101 @@ export interface InternalQueueOptions {
 /**
  * Represents all attributes of a queue entity
  */
-export interface QueueDetails extends QueueOptions {
+export interface QueueDetails {
   /**
    * Name of the queue
    */
-  queueName?: string;
+  queueName: string;
+
+  /**
+   * Determines the amount of time in seconds in which a message should be locked for processing by a receiver. After this period, the message is unlocked and available for consumption by the next receiver. Settable only at queue creation time.
+   */
+  lockDuration: string;
+
+  /**
+   * The entity's size in bytes.
+   *
+   */
+  sizeInBytes: number;
+
+  /**
+   * Specifies the maximum queue size in megabytes. Any attempt to enqueue a message that will cause the queue to exceed this value will fail.
+   */
+  maxSizeInMegabytes: number;
+
+  /**
+   * The entity's message count.
+   *
+   */
+  messageCount: number;
+
+  /**
+   * Depending on whether DeadLettering is enabled, a message is automatically moved to the DeadLetterQueue or deleted if it has been stored in the queue for longer than the specified time. This value is overwritten by a TTL specified on the message if and only if the message TTL is smaller than the TTL set on the queue. This value is immutable after the Queue has been created.
+   */
+  defaultMessageTimeToLive: string;
+
+  /**
+   * Specifies the time span during which the Service Bus detects message duplication.
+   */
+  duplicateDetectionHistoryTimeWindow: string;
+
+  /**
+   * Entity to forward deadlettered messages to
+   *
+   */
+  forwardDeadLetteredMessagesTo?: string;
+
+  /**
+   * Max idle time before entity is deleted
+   *
+   */
+  autoDeleteOnIdle: string;
+
+  /**
+   * The maximum delivery count.
```

Comment:
Can you add more description here?

Reply:
A rule of thumb to use is to have description that adds to the name of the property. If the description does not tell more than what the property name does, then it is redundant and has scope for improvement.

Reply:
Updated